contracts/test/integration/vip/vip13.sol

```diff
@@ -0,0 +1,484 @@
+                    MainnetAddresses.COMPOUND_USDC_PCV_DEPOSIT
+                ),
+                description: "Connect USDC deposit to PSM in ERC20 Allocator"
+            })
+        );
+        proposal.push(
+            ITimelockSimulation.action({
+                value: 0,
+                target: MainnetAddresses.ERC20ALLOCATOR,
+                arguments: abi.encodeWithSignature(
+                    "connectDeposit(address,address)",
+                    address(voltV2DaiPriceBoundPSM),
+                    MainnetAddresses.COMPOUND_DAI_PCV_DEPOSIT
+                ),
+                description: "Connect DAI deposit to PSM in ERC20 Allocator"
+            })
+        );
+
+        proposal.push(
+            ITimelockSimulation.action({
+                value: 0,
+                target: MainnetAddresses.PCV_GUARDIAN,
+                arguments: abi.encodeWithSignature(
+                    "addWhitelistAddresses(address[])",
+                    toWhitelist
+                ),
+                description: "Add new DAI, and USDC PSMs to PCV Guardian whitelist"
+            })
+        );
+        prop = proposal;
+    }
+
+    function mainnetSetup() public override {
+        // new pcv guardian to be deployed with safe address as timelock
+        uint256 governorVoltBalanceBeforeUsdc = IERC20(MainnetAddresses.VOLT)
+            .balanceOf(MainnetAddresses.GOVERNOR);
+
+        vm.startPrank(MainnetAddresses.GOVERNOR);
+
+        PCVGuardian(MainnetAddresses.PCV_GUARDIAN)
+            .withdrawAllERC20ToSafeAddress(
+                MainnetAddresses.VOLT_USDC_PSM,
+                MainnetAddresses.VOLT
+            );
+
+        uint256 governorVoltBalanceAfterUsdc = IERC20(MainnetAddresses.VOLT)
+            .balanceOf(MainnetAddresses.GOVERNOR);
+
+        voltInUsdcPSM =
+            governorVoltBalanceAfterUsdc -
+            governorVoltBalanceBeforeUsdc;
+
+        uint256 governorVoltBalanceBeforeDai = IERC20(MainnetAddresses.VOLT)
+            .balanceOf(MainnetAddresses.GOVERNOR);
+
+        PCVGuardian(MainnetAddresses.PCV_GUARDIAN)
+            .withdrawAllERC20ToSafeAddress(
+                MainnetAddresses.VOLT_DAI_PSM,
+                MainnetAddresses.VOLT
+            );
+
+        uint256 governorVoltBalanceAfterDai = IERC20(MainnetAddresses.VOLT)
+            .balanceOf(MainnetAddresses.GOVERNOR);
+
+        voltInDaiPSM =
+            governorVoltBalanceAfterDai -
+            governorVoltBalanceBeforeDai;
+
+        IERC20(MainnetAddresses.VOLT).transfer(
+            MainnetAddresses.TIMELOCK_CONTROLLER,
+            voltInDaiPSM + voltInUsdcPSM
+        );
+        vm.stopPrank();
+    }
+
+    function mainnetValidate() public override {
+        // Volt Token Validations
+        assertEq(voltV2.decimals(), 18);
+        assertEq(voltV2.symbol(), "VOLT");
+        assertEq(voltV2.name(), "Volt");
+        assertEq(voltV2.totalSupply(), voltInUsdcPSM + voltInDaiPSM);
```

Comment:
this total supply should equal the current volt v1 total supply to allow for all users to swap over.